ocs_ci/deployment/deployment.py

```diff
@@ -476,6 +481,84 @@ def deployment_with_ui(self):
             log_suffix="ui-deployment"
         )
 
+    def deploy_with_independent_mode(self):
+        """
+        This function handles the deployment of OCS on
+        external/indpendent RHCS cluster
+
+        """
+        live_deployment = config.DEPLOYMENT.get('live_deployment')
+        logger.info("Deploying OCS with independent mode RHCS")
+        logger.info("Creating namespace and operator group")
+        run_cmd(f"oc create -f {constants.OLM_YAML}")
+        if not live_deployment:
+            self.create_ocs_operator_source()
+        self.subscribe_ocs()
+        operator_selector = get_selector_for_ocs_operator()
+        package_manifest = PackageManifest(
+            resource_name=defaults.OCS_OPERATOR_NAME,
+            selector=operator_selector,
+        )
+        package_manifest.wait_for_resource(timeout=300)
+        channel = config.DEPLOYMENT.get('ocs_csv_channel')
+        csv_name = package_manifest.get_current_csv(channel=channel)
+        csv = CSV(resource_name=csv_name, namespace=self.namespace)
+        csv.wait_for_phase("Succeeded", timeout=720)
+
+        # Create secret for independent cluster
+        secret_data = templating.load_yaml(
+            constants.INDEPENDENT_CLUSTER_SECRET_YAML
+        )
+        external_cluster_details = config.INDEPENDENT_MODE.get(
+            'external_cluster_details',
+            ''
+        )
+        if not external_cluster_details:
+            raise ExternalClusterDetailsException(
+                "No external cluster data found"
+            )
+        secret_data['data']['external_cluster_details'] = (
+            external_cluster_details
+        )
+        secret_data_yaml = tempfile.NamedTemporaryFile(
+            mode='w+', prefix='external_cluster_secret', delete=False
+        )
+        templating.dump_data_to_temp_yaml(
+            secret_data, secret_data_yaml.name
+        )
+        logger.info("Creating external cluster secret")
+        run_cmd(f"oc create -f {secret_data_yaml.name}")
+
+        cluster_data = templating.load_yaml(
+            constants.INDEPENDENT_STORAGE_CLUSTER_YAML
+        )
+        cluster_data['metadata']['name'] = config.ENV_DATA[
+            'storage_cluster_name'
+        ]
+        cluster_data_yaml = tempfile.NamedTemporaryFile(
+            mode='w+', prefix='external_cluster_storage', delete=False
+        )
+        templating.dump_data_to_temp_yaml(
+            cluster_data, cluster_data_yaml.name
+        )
+        run_cmd(f"oc create -f {cluster_data_yaml.name}", timeout=2400)
+        self.independent_post_deploy_validation()
+        setup_ceph_toolbox()
+
+    def independent_post_deploy_validation(self):
+        """
+        This function validates successful deployment of OCS
+        in independent mode, some of the steps overlaps with
+        converged mode
+
+        """
+        # TODO: 1. HEALTH_OK + connected state from cephcluster
```

Comment:
In our opinion, we would need check for both "Phase" and "Health" and handle accordingly

$ oc get cephcluster
NAME                                         DATADIRHOSTPATH   MONCOUNT   AGE   PHASE       MESSAGE                          HEALTH
ocs-independent-storagecluster-cephcluster                                14d   Connected   Cluster connected successfully   HEALTH_OK


for **Phase**-  we need to cover https://bugzilla.redhat.com/show_bug.cgi?id=1851009
For **Health**: Will we fail the deployment if it is not HEALTH_OK ?
**reason**: It may happen that the RHCS cluster is in Warn state due to object counts, Slow IO on MDS etc.. Will we fail the deployment even for these issues ?

Reply:
ack, ll do it.